app/main.go

```diff
@@ -87,106 +50,81 @@ var (
 	Commit = ""
 )
 
-// add toml config file support and sample toml file
-
-func newSpacemeshApp() *SpacemeshApp {
-	app := cli.NewApp()
-	app.Name = filepath.Base(os.Args[0])
-	app.Author = config.AppAuthor
-	app.Email = config.AppAuthorEmail
-	app.Version = Version
-	if len(Commit) > 8 {
-		app.Version += " " + Commit[:8]
-	}
-	app.Usage = config.AppUsage
-	app.HideVersion = true
-	app.Copyright = config.AppCopyrightNotice
-	app.Commands = []cli.Command{
-		config.NewVersionCommand(Version, Branch, Commit),
-		// add all other commands here
-	}
-
-	app.Flags = append(app.Flags, appFlags...)
-	app.Flags = append(app.Flags, nodeFlags...)
-	app.Flags = append(app.Flags, apiFlags...)
-
-	sort.Sort(cli.FlagsByName(app.Flags))
-
-	sma := &SpacemeshApp{app, nil, make(chan bool, 1), nil, nil}
+// ParseConfig unmarshal config file into struct
+func ParseConfig() (*cfg.Config, error) {
+	conf := cfg.DefaultConfig()
 
-	// setup callbacks
-	app.Before = sma.before
-	app.Action = sma.startSpacemeshNode
-	app.After = sma.cleanup
+	err := viper.Unmarshal(&conf)
 
-	return sma
-}
+	if err != nil {
+		fmt.Printf("Failed to parse config ")
```

Comment:
log should still work also here since we init it inside log's `init()` func please check.
Also - please if no config file or config file is messed up it should fallback to default and throw notice, and not exit like now.